app-games/src/main/java/com/aptoide/android/aptoidegames/gamegenie/presentation/GameGenieAppViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.aptoide.android.aptoidegames.gamegenie.presentation
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import cm.aptoide.pt.feature_apps.data.App
+import cm.aptoide.pt.feature_apps.presentation.AppUiState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import timber.log.Timber
+import java.io.IOException
+
+class GameGenieAppViewModel(
+  private val app: App,
+) : ViewModel() {
+
+  private val viewModelState = MutableStateFlow<AppUiState>(AppUiState.Loading)
+
+  val uiState = viewModelState
+    .stateIn(
+      viewModelScope,
+      SharingStarted.Eagerly,
+      viewModelState.value
+    )
+
+  init {
+    reload()
+  }
+
+  fun reload() {
+    viewModelScope.launch {
```

Comment:
this viewmodel is not doing anything, it is only emitting the App after a small loading